distributed/comm/tcp.py

```diff
@@ -211,34 +211,47 @@ def closed(self):
         return self.stream is None or self.stream.closed()
 
 
+def _protocol_name(connection_kwargs):
+    """
+    The communication protocol that this connection is using
```

Comment:
I think this is really the wrong way of doing it. If the user asked for `tcp`, it should be given a TCP connection, regardless of what `connection_kwargs` is. Conversely, asking for `tls` should always give a TCP connection (and perhaps bail out if no suitable cert and private key is given, at least for a server connection).